client/gutenberg/editor/calypsoify-iframe.tsx

```diff
@@ -160,31 +160,20 @@ class CalypsoifyIframe extends Component<
 	mediaCancelPort: MessagePort | null = null;
 	revisionsPort: MessagePort | null = null;
 	checkoutPort: MessagePort | null = null;
-	successfulIframeLoad = false;
-	waitForIframeToInit: ReturnType< typeof setInterval > | undefined = undefined;
-	waitForIframeToLoad: ReturnType< typeof setTimeout > | undefined = undefined;
 
 	componentDidMount() {
 		window.addEventListener( 'message', this.onMessage, false );
 
-		const isDesktop = config.isEnabled( 'desktop' );
-		// If the iframe fails to load for some reson, eg. an unexpected auth loop, this timeout
-		// provides a redirect to wpadmin for web users - this should now be a rare occurance with
-		// a 3rd party cookie auth issue fix in place https://github.com/Automattic/jetpack/pull/16167
-		this.waitForIframeToInit = setInterval( () => {
+		// If the iframe itself fails to load for some reason, eg. an unexpected
+		// auth loop, we need to exit to wp-admin without the iFrame. Therefore,
+		// once we start loading the iFrame, we set the editor redirect timer to
+		// 25s. This timer is cleared when iFrame.onLoad is called.
+		const waitForIframeToInit = setInterval( () => {
```

Comment:
Would `onerror` and `onabort` event handlers catch these faster than a timer?

Reply:
Good thought! I was looking for something like that but didn't find anything. Will investigate

Reply:
@simison I'm no iframe expert, but my best understanding of the situation is that those don't really work for iframes, particularly cross-origin iframes.